includes/class-amp-story-post-type.php

```diff
@@ -1368,4 +1371,48 @@ public static function remove_title_from_embed( $output, $post ) {
 
 		return preg_replace( '/<blockquote class="wp-embedded-content">.*?<\/blockquote>/', '', $output );
 	}
+
+	/**
+	 * Changes attributes of the AMP Story embed <iframe>.
+	 *
+	 * The embed typically appears in the editor in an <iframe>, and without an <iframe> on the front-end.
+	 * The height of the <iframe> isn't enough to display the full story, so this increases it.
+	 * Also, this removes attributes that AMP does not allow for the <amp-iframe>, like marginwidth.
+	 *
+	 * @param string  $output The embed output.
+	 * @param WP_Post $post The post for the embed.
+	 * @return string The filtered embed output.
+	 */
+	public static function change_embed_iframe_attributes( $output, $post ) {
+		if ( self::POST_TYPE_SLUG !== get_post_type( $post ) ) {
+			return $output;
+		}
+
+		preg_match( '/<iframe sandbox="allow-scripts" [^>]*>/', $output, $matches );
+		if ( ! $matches ) {
+			return $output;
+		}
+
+		$original_iframe = $matches[0];
+		$new_iframe      = preg_replace(
+			'/(?<=\sheight=")\w+(?=")/',
+			strval( ( self::STORY_LARGE_IMAGE_DIMENSION / 2 ) + 4 ),
+			$original_iframe
+		);
+
+		// Remove attributes that AMP doesn't allow for the <amp-iframe>.
+		if ( is_amp_endpoint() ) {
+			$new_iframe = preg_replace(
+				array(
+					'/\ssecurity="[\w]+"/',
+					'/\smarginwidth="[\w]+"/',
+					'/\smarginheight="[\w]+"/',
+				),
```

Comment:
This array can be replaced with a single string:

`/\s(security|marginwidth|marginheight)="[^"]*"/`

Reply:
Thanks, good idea. At the moment, it looks like the logic to remove these attributes will be removed.